Bottom: 68b2eea1d6a6b364c55c032e8b60e053d2ec602b
Top:    0b3373d787027a124aa23ea01f8b1d011e3023da
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-08-31 11:23:36 +0800

Refresh of vhost_net-basic-batch

---

diff --git a/drivers/vhost/vhost.c b/drivers/vhost/vhost.c
index 4bcd096..daeec47 100644
--- a/drivers/vhost/vhost.c
+++ b/drivers/vhost/vhost.c
@@ -2490,27 +2490,6 @@ int vhost_prefetch_desc_indices(struct vhost_virtqueue *vq,
 	total = vq->avail_idx - vq->last_avail_idx;
 	ret = total = min(total, num);
 
-#if 0
-	/* FIXME: into heads directly */
-	while (total) {
-		/* FIXME: use u16 for vq->num */
-		copied = min((u16)(vq->num - last_avail_idx), total);
-		ret2 = vhost_copy_from_user(vq, &indices[ret - total],
-					    &vq->avail->ring[last_avail_idx],
-					    copied * sizeof *indices);
-		if (unlikely(ret2)) {
-			vq_err(vq, "Failed to get descriptors\n");
-			return -EFAULT;
-		}
-
-		last_avail_idx = 0;
-		total -= copied;
-	}
-
-	if (!heads)
-		goto out;
-#endif
-
 	for (i = 0; i < ret; i++) {
 		ret2 = vhost_get_avail(vq, heads[i].id,
 				      &vq->avail->ring[last_avail_idx]);
